presto-main/src/main/java/com/facebook/presto/sql/gen/IsDistinctFromCodeGenerator.java

```diff
@@ -58,6 +59,22 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
                 .comment("equals(%s, %s)", leftType, rightType)
                 .append(invoke(binding, operator.getSignature()));
 
+        ByteCodeNode neitherSideIsNull;
+        if (leftType instanceof UnknownType || rightType instanceof UnknownType) {
+            // the generated block should be unreachable. However, a boolean need to be pushed to balance the stack
+            neitherSideIsNull = new Block()
+                    .comment("unreachable code")
+                    .push(true);
```

Comment:
this is unreachable, so it doesn't matter. But this should probably be `false` rather than `true` since no value of `UnknownType` can be distinct from another